concordium-node/src/macos_log.rs

```diff
@@ -111,6 +111,7 @@ pub struct MacOsLogger {
 }
 
 impl MacOsLogger {
+    #[allow(dead_code)]
```

Comment:
I'm not sure about this.

Reply:
I tried it, but it does not 🤔

Reply:
This entire file should be removed. I thought I did it. It is not longer used which is why you are getting these warnings.

There is now a separate crate used by both the node and the collector that has this contents.

So can you please
- delete this file
- remove 
  ```
#[cfg(target_os = "macos")]
mod macos_log;
```
from lib.rs
- test that it still builds fine on the mac. I don't see this imported anywhere so it should.